### PR TITLE
Enable Fast CRC32 for Win64

### PR DESCRIPTION
**Pull request by min-mwei**
Currently the fast crc32 path is not enabled on Windows. I am trying to enable it here, hopefully, with the minimum impact to the existing code structure.

### CODE REVIEW

**Comment by siying** [comment]
@min-mwei @yuslepukhin Before the Pull Request, Fast CRC is not supported in Windows?

**Comment by min-mwei** [comment]
That's right, for the versions I started to use RocksDB since 5.0. The summary in the LOG started false/0. In some of my tests recently, it seems to have ~7% overall perf impact. So I think it should be enabled.

**Comment by siying** [comment]
@yuslepukhin can you comment on @min-mwei 's claim?

**Comment by yuslepukhin** [comment]
@siying Yes, Min is correct. I just spoke to him and this was completely overlooked. We agreed that he will run all the tests locally not just those in the CI build and then he will let you know when it is ready. I also have a minor comment there for the use of __cpuidex

File: util/crc32c.cc [hunk 1]
```diff
@@ -326,6 +338,11 @@ static inline void Fast_CRC32(uint64_t* l, uint8_t const **p) {
   *l = _mm_crc32_u32(static_cast<unsigned int>(*l), LE_LOAD32(*p));
   *p += 4;
 #endif
+#elif defined(_WIN64)
+#ifdef __AVX2__
+  *l = _mm_crc32_u64(*l, LE_LOAD64(*p));
+  *p += 8;
+#endif
```
Comment (siying) [inline util/crc32c.cc]: Do we need to do something if defined(_WIN64) and !__AVX2__?

**Comment by min-mwei** [comment]
Per Dmitri's instructions, all tests, db_test, db_test2 passed with the exception of one rate limit test:
  WARNING:  1 FAILED TEST
  WARNING: db_test_je.exe : write rate ratio = 0.64, expected 0.7
  WARNING: At line:3 char:5
  WARNING: +     &$exe --gtest_filter=$test > $log 2>&1
  WARNING: +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  WARNING:     + CategoryInfo          : NotSpecified: (write rate ratio = 0.64, expected 0.7:String) [],  RemoteException

This test passed when I run it manually.

**Comment by siying** [comment]
@yuslepukhin @min-mwei is it ready to merge?

**Comment by min-mwei** [comment]
I run the tests with AVX2 on and off, with no major issue (except the rate limit test on both).  It's a go from my side.

**Comment by siying** [comment]
If @yuslepukhin has no objection, I'm going to merge it.

**Comment by yuslepukhin** [comment]
@siying @min-mwei I think we might improve on the Cmake file. One more argument is easy to miss and get a build without this improvement. I would say we want to either make a it a default on 64-bit OR better yet, automatically detect. I would check CMake capabilities first to see if it already has a built-in mechanism to answer the question.

**Comment by min-mwei** [comment]
I just flipped the USE_AVX2 on by default. 

From my limited Web search, there does not seem to be an easy or more portable way to detect CPU info with CMake. We could lift the IsSSE supported code snippet out into a standalone tool for CMake to use. This might be an overkill. The other thing is that, people might compile code on an old machine for a cloud deployment which has new processors.  

Let me know, I am open to suggestions.

**Comment by yuslepukhin** [comment]
@siying I guess you can merge, it is mostly about people who build w/o that support.

**Comment by siying** [comment]
Thank you for your contribution!

**Pull request closed without merging**